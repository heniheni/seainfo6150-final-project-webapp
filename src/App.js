import { isEmpty } from "lodash";
import React, { useEffect, useState } from "react";
import { Switch, Route, Link } from "react-router-dom";
import styles from "./App.module.css";
import Home from "./Home/Home.jsx";

import Bar from "./Bar/Bar.jsx";
import Baz from "./Baz/Baz.jsx";
import Error from "./Error/Error.jsx";
import ipl from "../src/ipl.png";
import Teams from "./Teams/Teams.jsx";
import Form from "./FeedBackForm/FeedBackForm.jsx";
import TeamDetail from "./Teams/TeamDetail.jsx";
import ContactUs from "./ContactUs/ContactUs.jsx";
import News from "./News/News.jsx";
import NewsDetail from "./News/NewsDetail.jsx";
import AboutUs from "./AboutUs/AboutUs.jsx";

// here is some external content. look at the /baz route below
// to see how this content is passed down to the components via props
const externalContent = {
  id: "article-1",
  title: "An Article",
  author: "April Bingham",
  text: "Some text in the article",
};

function App() {
  const [fetchedData, setFetchedData] = useState();
  const [fetchedNews, setFetchedNews] = useState();

  useEffect(() => {
    const fetchData = async () => {
      // performs a GET request
      const response = await fetch("http://localhost:8000/teams");
      const responseJson = await response.json();
      setFetchedData(responseJson);
    };

    if (isEmpty(fetchedData)) {
      fetchData();
    }

    const fetchNews = async () => {
      // performs a GET request
      const responseNews = await fetch("http://localhost:8000/News");
      const responseNewsJson = await responseNews.json();
      setFetchedNews(responseNewsJson);
    };

    if (isEmpty(fetchedNews)) {
      fetchNews();
    }
  }, [fetchedData, fetchedNews]);

  return (
    <div className={styles.mainContainer}>
      <div className={styles.headerContainer}>
        <header>
          <nav className={styles.navContainer}>
          <Link to="/"><img className={styles.logoBox} src={ipl} alt="IPL Player Cricket"/></Link>

            {/* <div  className={styles.menuContainer}> */}
            <ul className={styles.listControl}>
              {/* these links should show you how to connect up a link to a specific route */}
              <li>
                <Link className={styles.stylingLink} to="/">
                  Home
                </Link>
              </li>
              <li>
                <Link className={styles.stylingLink} to="/Teams">
                  Teams
                </Link>
              </li>
              <li>
                <Link className={styles.stylingLink} to="/News">
                  News
                </Link>
              </li>
              <li>
                <Link className={styles.stylingLink} to="/AboutUs">
                  About Us
                </Link>
              </li>
              <li>
                <Link className={styles.stylingLink} to="/form">
                  Feedback
                </Link>
              </li>

              <li>
                <Link className={styles.stylingLink} to="/ContactUs">
                  Contact Us
                </Link>
              </li>
            </ul>
            {/* </div> */}
          </nav>
        </header>
      </div>

      {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
      <div className={styles.centerArea}>
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/aboutus" exact component={AboutUs} />

          <Route
            exact
            path={`/newsDetail/:newsId`}
            render={({ match }) => {
              return fetchedNews ? (
                <NewsDetail newsDetail={fetchedNews[match.params.newsId]} />
              ) : null;
            }}
          ></Route>

          {/* <Route path="/foo" exact component={Foo} /> */}
          <Route path="/form" exact component={Form} />

          <Route path="/news" exact component={News}>
            {isEmpty(fetchedNews) ? (
              "Team Data Loading..."
            ) : (
              <News news={Object.values(fetchedNews)} />
            )}
          </Route>

          <Route path="/contactus" exact component={ContactUs} />

          <Route path="/teams" exact component={Teams}>
            {isEmpty(fetchedData) ? (
              "Team Data Loading..."
            ) : (
              <Teams teams={Object.values(fetchedData)} />
            )}
          </Route>
          {/* passing parameters via a route path */}
          <Route
            path="/bar/:categoryId/:productId"
            exact
            render={({ match }) => (
              // getting the parameters from the url and passing
              // down to the component as props

              <Bar
                categoryId={match.params.categoryId}
                productId={match.params.productId}
              />
            )}
          />
          <Route
            path="/baz"
            exact
            render={() => <Baz content={externalContent} />}
          />

          <Route
            exact
            path={`/TeamDetail/:teamReference`}
            render={({ match }) => {
              // getting the parameters from the url and passing
              // down to the component as props
              return fetchedData ? (
                <TeamDetail team={fetchedData[match.params.teamReference]} />
              ) : null;
            }}
          />

          <Route component={Error} />
        </Switch>
      </div>
      <footer>
        Developed By: <i>Heni Bhungalia</i> 
        <Link className={styles.footerLink} to="/ContactUs">
          Contact Us
        </Link>
      </footer>
    </div>
  );
}

export default App;
