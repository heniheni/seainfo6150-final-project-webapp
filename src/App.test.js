import React from "react";
import App from "./App";


describe("App tests", () => {

    it("renders correctly", () => {
      const { container } = render(<App />);
      expect(container).toMatchSnapshot();
    });
  });