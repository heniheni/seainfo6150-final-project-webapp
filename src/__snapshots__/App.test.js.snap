// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders correctly 1`] = `
<div>
  <div
    class="mainContainer"
  >
    <div
      class="headerContainer"
    >
      <header>
        <nav
          class="navContainer"
        >
          <a
            href="/"
          >
            <img
              alt="IPL Player Cricket"
              class="logoBox"
              src="ipl.png"
            />
          </a>
          <ul
            class="listControl"
          >
            <li>
              <a
                class="stylingLink"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class="stylingLink"
                href="/Teams"
              >
                Teams
              </a>
            </li>
            <li>
              <a
                class="stylingLink"
                href="/News"
              >
                News
              </a>
            </li>
            <li>
              <a
                class="stylingLink"
                href="/AboutUs"
              >
                About Us
              </a>
            </li>
            <li>
              <a
                class="stylingLink"
                href="/form"
              >
                Feedback
              </a>
            </li>
            <li>
              <a
                class="stylingLink"
                href="/ContactUs"
              >
                Contact Us
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
    <div
      class="centerArea"
    >
      <div>
        <div>
          <img
            alt=""
            class="ImageSlider"
            src="one.jpeg"
          />
           
          <br />
          <div
            class="directionalButtons"
          >
            <button
              class="previousButton"
            >
              &lt;
            </button>
            <button
              class="forwardButton"
            >
              &gt;
            </button>
          </div>
        </div>
      </div>
      <div>
        <h1
          class="headerStyling"
        >
          2021 Leaders
        </h1>
      </div>
      <div
        class="leadersContainer"
      >
        <div
          class="playerOneContainer"
        >
          <img
            alt="Sport Player First"
            src="https://static.iplt20.com/players/210/41.png"
          />
          <div
            class="awardOneContainer"
          >
            <strong
              class="awardOne"
            >
              Orange Cap
            </strong>
          </div>
          <div
            class="playerOneInfo"
          >
            <div>
              Sikhar
            </div>
            <div>
              <strong
                class="stylingName"
              >
                Dhawan
              </strong>
            </div>
            <div>
              <strong
                class="stylingDigit"
              >
                231
              </strong>
            </div>
            <div>
              <small
                class="stylingText"
              >
                Runs
              </small>
            </div>
          </div>
        </div>
        <div
          class="playerTwoContainer"
        >
          <img
            alt="Sport Player Second"
            src="https://static.iplt20.com/players/210/157.png"
          />
          <div
            class="awardTwoContainer"
          >
            <strong
              class="awardTwo"
            >
              Purple Cap
            </strong>
          </div>
          <div
            class="playerOneInfo"
          >
            <div>
              Harshal
            </div>
            <div>
              <strong
                class="stylingName"
              >
                Patel
              </strong>
            </div>
            <div>
              <strong
                class="stylingDigit"
              >
                12
              </strong>
            </div>
            <div>
              <small
                class="stylingText"
              >
                Wickets
              </small>
            </div>
          </div>
        </div>
        <div
          class="playerThreeContainer"
        >
          <img
            alt="Sport Player Third"
            src="https://static.iplt20.com/players/210/258.png"
          />
          <div
            class="awardThreeContainer"
          >
            <strong
              class="awardThree"
            >
              Highest Score
            </strong>
          </div>
          <div
            class="playerOneInfo"
          >
            <div>
              Sanju
            </div>
            <div>
              <strong
                class="stylingName"
              >
                Samson
              </strong>
            </div>
            <div>
              <strong
                class="stylingDigit"
              >
                119
              </strong>
            </div>
            <div>
              <small
                class="stylingText"
              >
                Score
              </small>
            </div>
          </div>
        </div>
        <div
          class="playerFourContainer"
        >
          <img
            alt="Sport Player Fourth"
            src="https://static.iplt20.com/players/210/177.png"
          />
          <div
            class="awardFourContainer"
          >
            <strong
              class="awardFourStyling"
            >
              Best Bowling Figures
            </strong>
          </div>
          <div
            class="playerOneInfo"
          >
            <div>
              Andre
            </div>
            <div>
              <strong
                class="stylingName"
              >
                Russell
              </strong>
            </div>
            <div>
              <strong
                class="stylingDigit"
              >
                5/15
              </strong>
            </div>
            <div>
              <small
                class="stylingText"
              >
                Wickets
              </small>
            </div>
          </div>
        </div>
        <div
          class="playerFiveContainer"
        >
          <img
            alt="Sport Player Fifth"
            src="https://static.iplt20.com/players/210/41.png"
          />
          <div
            class="awardFiveContainer"
          >
            <strong
              class="awardFiveStyling"
            >
              Upstox Most Valuable Player
            </strong>
          </div>
          <div
            class="playerFiveInfo"
          >
            <div>
              Shikhar
            </div>
            <div>
              <strong
                class="stylingName"
              >
                Dhawan
              </strong>
            </div>
            <div>
              <strong
                class="stylingDigit"
              >
                107.5
              </strong>
            </div>
            <div>
              <small
                class="stylingText"
              >
                pts
              </small>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h1
          class="pointTableHeaderStyling"
        >
          Point Table
        </h1>
      </div>
      <div
        class="tableContainer"
      >
        <table
          class="scheduleTable"
        >
          <thead>
            <tr
              class="tableFirstRow"
            >
              <th />
              <th>
                Team
              </th>
              <th>
                Pld
              </th>
              <th>
                Won
              </th>
              <th>
                Lost
              </th>
              <th>
                Tied
              </th>
              <th>
                N/R
              </th>
              <th>
                Net RR
              </th>
              <th>
                For
              </th>
              <th>
                Against
              </th>
              <th>
                Pts
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td
                class="shadowTable"
              >
                Delhi Capitals
              </td>
              <td>
                1
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                +0.779
              </td>
              <td>
                190/18.4
              </td>
              <td>
                188/20
              </td>
              <td>
                2
              </td>
            </tr>
            <tr>
              <td>
                2
              </td>
              <td
                class="shadowTable"
              >
                Kolkata Knight Riders
              </td>
              <td>
                1
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                +0.500
              </td>
              <td>
                187/20
              </td>
              <td>
                177/20
              </td>
              <td>
                2
              </td>
            </tr>
            <tr>
              <td>
                3
              </td>
              <td
                class="shadowTable"
              >
                Punjab Kings
              </td>
              <td>
                1
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                +0.200
              </td>
              <td>
                221/20
              </td>
              <td>
                217/20
              </td>
              <td>
                2
              </td>
            </tr>
            <tr>
              <td>
                4
              </td>
              <td
                class="shadowTable"
              >
                Royal Challengers Bangalore
              </td>
              <td>
                1
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                +0.050
              </td>
              <td>
                160/20
              </td>
              <td>
                159/20
              </td>
              <td>
                2
              </td>
            </tr>
            <tr>
              <td>
                5
              </td>
              <td
                class="shadowTable"
              >
                Mumbai Indians
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                -0.050
              </td>
              <td>
                159/20
              </td>
              <td>
                160/20
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                6
              </td>
              <td
                class="shadowTable"
              >
                Rajasthan Royals
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                -0.200
              </td>
              <td>
                217/20
              </td>
              <td>
                221/20
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                7
              </td>
              <td
                class="shadowTable"
              >
                Sunrisers Hyderabad
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                -0.500
              </td>
              <td>
                177/20
              </td>
              <td>
                187/20
              </td>
              <td>
                0
              </td>
            </tr>
            <tr>
              <td>
                8
              </td>
              <td
                class="shadowTable"
              >
                Chennai Super Kings
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                1
              </td>
              <td>
                0
              </td>
              <td>
                0
              </td>
              <td>
                -0.779
              </td>
              <td>
                188/20
              </td>
              <td>
                190/18.4
              </td>
              <td>
                0
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <footer>
      Developed By: 
      <i>
        Heni Bhungalia
      </i>
      <a
        class="footerLink"
        href="/ContactUs"
      >
        Contact Us
      </a>
    </footer>
  </div>
</div>
`;
